app/src/main/java/com/example/habitsmasher/FollowUserDialog.java

```diff
@@ -68,10 +72,35 @@ public View onCreateView(@NonNull LayoutInflater inflater,
         final Dialog followUserDialog = getDialog();
 
         // Attach UI elements
-        _userToFollow = view.findViewById(R.id.user_search_text);
+        _userToFollowACTV = view.findViewById(R.id.auto_complete_text_view);
         Button cancelButton = view.findViewById(R.id.cancel_user_follow);
         Button followButton = view.findViewById(R.id.follow_user_button);
 
+        _usernames = new ArrayList<String>();
+
+        _db.collection(USERS_COLLECTION_PATH)
+                .get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                if (task.isSuccessful()) {
+                    for (QueryDocumentSnapshot document : task.getResult()) {
+                        if (document.exists()) {
+                            if (document.get(USERNAME_FIELD).toString().contains(_userToFollowACTV.getText().toString())) {
+                                Log.d(TAG, "Username exists");
+                                _usernames.add(document.get(USERNAME_FIELD).toString());
```

Comment:
if username in database contains what the user searches, add to the ArrayList

Reply:
Right now, when we query the database for users' usernames, it will not be case sensitive to what the user enters. Attached a screenshot to show it works correctly.
![CMPUT301HabitSmasher#142 7](https://user-images.githubusercontent.com/77360509/143780912-a9221164-188a-48df-b758-56db6e0efef3.PNG)